src/sentry.cr

```diff
@@ -1,6 +1,99 @@
+require "yaml"
+
 module Sentry
   FILE_TIMESTAMPS = {} of String => String # {file => timestamp}
 
+  class Config
+    # `process_name` is set as a class property so that it can be inferred from
+    # the `shard.yml` in the project directory.
+    class_property! process_name : String?
+
+    YAML.mapping(
+      name: {
+        type: String?,
+        getter: false,
+        default: nil
+      },
+      info: {
+        type: Bool,
+        default: false
+      },
+      build: {
+        type: String?,
+        getter: false,
+        default: nil
+      },
+      build_args: {
+        type: String,
+        getter: false,
+        default: ""
+      },
+      run: {
+        type: String?,
+        getter: false,
+        default: nil
+      },
+      run_args: {
+        type: String,
+        getter: false,
+        default: ""
+      },
+      watch: {
+        type: Array(String),
+        default: ["./src/**/*.cr", "./src/**/*.ecr"]
+      }
+    )
+
+    def name
+      @name ||= self.class.process_name
+    end
+
+    def build
+      @build ||= "crystal build ./src/#{name}.cr"
```

Comment:
hmmm.... this bit is a breaking change. Originally, the "process name" was intended to be cosmetic, and only changed what the process was referred to when printed to the terminal. The actually default build and run commands were based off of the name in the shard.yml file. Now the process name argument will influence the build and run commands. Was that intentional?

Reply:
It was a bit unclear to me. When I originally looked at `sentry_cli.cr`, I saw that `process_name` was being interpolated into the string, so I naively assumed that it was meant to follow what was passed on the CLI, but looking again, that clearly isn't what the behavior was.

That was actually one of the harder things to implement when I was working on this, so I'm kind of glad it's not the intended behavior :) I'll revert that now.